src/__tests__/use-document/use-document.test.tsx

```diff
@@ -6,6 +6,16 @@ import { getPage } from '../../index';
 import CustomDocumentWithGIP_Page from './__fixtures__/custom-document-with-gip/pages/page';
```

Comment:
If you remove this import head will work correctly on first render.

Reply:
Good catch about `side-effect.ts`. It could definitely be the cause of initial render problem.

A few days ago I found out that it's possible resetting modules directly from userland code:
https://stackoverflow.com/questions/9210542/node-js-require-cache-possible-to-invalidate/16060619

Could it help us to work around these `isServer` flags?

BTW, with this project I learned that setting `isServer` flags at module level is probably an anti pattern. `typeof window === 'undefined'` should usually run at execution time.

Reply:
If it's possible to reset modules and we could get that to work that would be the best imo. Otherwise, we cannot control what imports user does in his tests and would need to have a FAQ about this.

Reply:
I'll give it a try.

I won't say this loud, but it seems that `react-hook-form` is the origin of our tests unpredictability. I might be doing something which conflicts with something else in `react-hook-form` which originates a race conditions.

Reply:
Jest re-implements `require` and doesn't provide `require.cache` by design.